core/camel-main/src/main/java/org/apache/camel/main/HttpServerConfigurationProperties.java

```diff
@@ -47,7 +47,10 @@ public class HttpServerConfigurationProperties implements BootstrapCloseable {
     private boolean uploadEnabled;
     private String uploadSourceDir;
 
+    private HttpServerAuthenticationConfigurationProperties authentication;
```

Comment:
Maybe make it final, for immutability and ease of maintenance?

Reply:
As far as I can understand, the problem is that with the current implementation of closing BootstrapCloseable properties, we need to set links to embedded properties to null so that they are eligible for gc. This is the same as in [org.apache.camel.main.MainConfigurationProperties](https://github.com/apache/camel/blob/main/core/camel-main/src/main/java/org/apache/camel/main/MainConfigurationProperties.java). So to make those final, we need to see if we can close *Properties.java objects in a different way, maybe this can be done in a separate issue/PR.

Reply:
AH, got it. That's fine, thanks for the explanation.